dynamax/linear_gaussian_ssm/inference.py

```diff
@@ -521,17 +549,17 @@ def _step(carry, args):
         smoothed_mean_next, smoothed_cov_next = carry
         t, filtered_mean, filtered_cov = args
 
-        # Get parameters and inputs for time index t
-        F, B, b, Q = _get_params(params, num_timesteps, t)[:4]
-        u = inputs[t]
+        # Get parameters and inputs for time index t + 1
+        F_next, B_next, b_next, Q_next = _get_params(params, num_timesteps, t + 1)[:4]
+        u_next = inputs[t + 1]
```

Comment:
A bug fix: calculation of the mean on the next time step requires parameters at the next time step.